wallet/wallet.c

```diff
@@ -2277,3 +2278,77 @@ struct bitcoin_txid *wallet_transactions_by_height(const tal_t *ctx,
 	return txids;
 }
 
+void wallet_channeltxs_add(struct wallet *w, struct channel *chan,
+			   const int type, const struct bitcoin_txid *txid,
+			   const u32 input_num, const u32 blockheight)
+{
+	sqlite3_stmt *stmt;
+	stmt = db_prepare(w->db, "INSERT INTO channeltxs ("
+			  "  channel_id"
+			  ", type"
+			  ", transaction_id"
+			  ", input_num"
+			  ", blockheight"
+			  ") VALUES (?, ?, ?, ?, ?);");
+	sqlite3_bind_int(stmt, 1, chan->dbid);
```

Comment:
Is the 'id' database field used at all?

Reply:
It's an autoincrement field that is used in the `SELECT` below to make sure we replay the events in the same order they were created, i.e., seen originally. This is just for added consistency should we have some dependency on the order (multiple transactions being confirmed in the same block).

Reply:
It is declared `INTEGER PRIMARY KEY`, but note, this implies it is equivalent to `ROWID` (an internal detail) under SQLITE.  In particular, if a row is deleted, it is possible for SQLITE to reuse that row when another row is inserted. We need an explicit `AUTOINCREMENT` in the declaration of `id` in `CREATE TABLE channeltxs`.

https://www.sqlite.org/autoinc.html

> If the largest ROWID is equal to the largest possible integer (9223372036854775807) then the database engine starts picking positive candidate ROWIDs at random until it finds one that is not previously used. If no unused ROWID can be found after a reasonable number of attempts, the insert operation fails with an SQLITE_FULL error.

Admittedly the condition of reaching 2^63 is extremely unlikely, and if we neglect that, we can generally think of `INTEGER PRIMARY KEY` as implicitly `AUTOINCREMENT`, but I think explicitness is a win here.